### PR TITLE
Implement model training functions and exporting metadata

### LINKED ISSUE #29: Implement model training functions
I will implement training the data given the different hyperparameters and provide the client with a machine learning model

### PR DESCRIPTION
**Pull request by LeoJung1**
## Purpose

This PR resolves #29 by implementing the machine learning model training functions and enable the interaction through API endpoint `POST /api/models/` given different hyperparameters and return a model to the user.

## Changes

- Implement two different models(Logistics, Linear) and get there training accuracy and validation accuracy
- Implement functions that export the model and the model metadata
- Connect the training function so that it would be trained on the requested parameters sent by the client

### CODE REVIEW

File: app/services/model_service.py [hunk 1]
```diff
@@ -74,3 +99,24 @@ def _prepare_dataset(
...
+            f.write(f"Number of Features : {model_metadata.num_features}\n")
+            f.write(f"Learning Rate : {model_metadata.learning_rate}\n")
+            f.write(f"K : {model_metadata.k}\n")
+    
+        f.close()
+
+
+    @staticmethod
+    def _save_model(model_id: str, model: RegressorMixin) -> None:
+        filepath = f"../data/models/model/model_{model_id}.pkl"
```
Comment (kyminbb) [inline app/services/model_service.py:119]: Same here!

File: app/services/model_service.py [hunk 2]
```diff
@@ -74,3 +99,24 @@ def _prepare_dataset(
         elif model_class == "logistic":
             y = df["G3"] >= 15.0
         return X, y
+    
+    @staticmethod
+    def _save_model_metadata(model_id: str, model_metadata: ModelMetadata) -> None:
+        filepath = f"../data/models/model_metadata/{model_metadata.model_class}_{model_id}.txt"
```
Comment (kyminbb) [inline app/services/model_service.py:105]: I think it would be safer to use `pathlib` to build the file path.

ex. 
```python
data_dir = Path().cwd().parent.joinpath("data")
filepath = data_dir.joinpath(f"models/model_metadata/{model_id}.pkl"))
```

**Review by kyminbb — CHANGES_REQUESTED** [review]

File: app/services/model_service.py (hunk 1 above)
Comment (LeoJung1) [reply app/services/model_service.py]: Fixed!

File: app/services/model_service.py (hunk 2 above)
Comment (LeoJung1) [reply app/services/model_service.py]: Fixed!

**Review by kyminbb — APPROVED** [review]
LGTM!!
